python/ray/_private/ray_logging/formatters.py

```diff
@@ -10,16 +11,24 @@ def generate_record_format_attrs(
     exclude_standard_attrs,
 ) -> dict:
     record_format_attrs = {}
+
+    # Python logging module starts to use `time.time_ns()` to generate `created`
+    # from Python 3.13 to avoid the precision loss caused by the float type. Here,
+    # we generate the `created` for the log record to support Python versions before 3.13.
+    ct = time.time_ns()
```

Comment:
Can we use `time.process_time_ns()` here instead? I think on macOS calling `time.time_ns()` will lost the last 3 digits

Reply:
`process_time_ns()` is the sum of the system and user CPU time of the current process. Is this what you are referring to?

Reply:
oops sorry, paste the wrong method, I meant `time.perf_counter_ns()` https://docs.python.org/3/library/time.html#time.perf_counter_ns

Reply:
hmm actually not `time.perf_counter_ns()` either 🤔

Reply:
but I'm referring, on MacOS, it looks like this
```
>>> import time
>>> time.time_ns()
1718650890402076000
>>> time.time_ns()
1718650892760831000
>>> time.time_ns()
1718650893345585000
>>> time.time_ns()
1718650893877456000
```
the last 3 digits are zeros

Reply:
ok so reading a bit more, we can probably get the last 3 digits by doing this https://stackoverflow.com/a/2394534, but I feel it's not worth the work since it's only an issue on MacOS. Feel free to keep the code as is.

Reply:
`perf_counter_ns` seems to be similar to `process_time_ns` but includes the sleep time. It seems not to be a replacement of `time_ns()`.

<img width="305" alt="Screenshot 2024-06-17 at 1 00 48 PM" src="https://github.com/ray-project/ray/assets/20109646/0339f2b7-0502-425e-8381-39bb4947619a">

Reply:
right, it's more of just a way to get the nanosecond precisions. but really at that point we might as well just substitute with 3 digit random numbers lol